libs/jsonforms-components/src/lib/Controls/FormStepper/FormStepperControl.tsx

```diff
@@ -157,6 +226,31 @@ export const FormStepper = ({
       setShowNextBtn(true);
     }
   }
+  function getAllRequiredFields() {
+    const requiredFields: string[] = [];
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    function findRequired(fields: any) {
+      if (fields.required && Array.isArray(fields.required)) {
+        fields.required.forEach((field: string) => {
+          requiredFields.push(field);
+        });
+      }
+
+      if (fields && fields.properties) {
+        Object.keys(fields.properties).forEach((key) => {
+          findRequired(fields.properties[key]);
+        });
+      } else if (fields.type === 'array' && fields.items && typeof fields.items === 'object') {
+        findRequired(fields.items);
+      }
+    }
+
+    findRequired(schema);
+
+    return requiredFields;
+  }
+
```

Comment:
Please move this function into a separate file.  Then you function's signature woudl be something like:
```
getAllRequiredFields(fields: Array<something>, schema: JsonSchema): Array<string> { ... }
```
Then, add another file for unit tests on this function.